atom/common/api/atom_api_clipboard.cc

```diff
@@ -50,9 +51,49 @@ std::string Read(const std::string& format_string,
   return data;
 }
 
+void Write(const std::vector<mate::Dictionary>& data,
+           mate::Arguments* args) {
+  ui::ScopedClipboardWriter writer(GetClipboardType(args));
+  for (uint32_t i = 0; i < data.size(); ++i) {
+    base::string16 text;
+    gfx::Image image;
+    std::string format;
+    ui::Clipboard::FormatType type;
+    if (data[i].Get("format", &format))
```

Comment:
:+1: , and since you can write data of a type only once under a writer instance, can the  final api be
```
clipboard.write({
  image: '..',
  text: '..',
  html: '..'
});
```